### ISSUE TITLE
Want add_project_arguments() and friends

### ISSUE THREAD

**Issue by tp-m** [mesonbuild/meson#979]
`add_global_arguments()` and friends are convenient, but unfortunately they don't work when the project is used as a submodule. As in: we will error out. This is unexpected and unfortunate, and guarding all `add_global_arguments()` with is_subproject checks is tedious.

Two options:
1. add `add_project_arguments()` and tell everyone to use that instead
2. make add_global_arguments() do magic inside a subproject, ie. magic that does the equivalent of 1.

Consensus seems to gravitate towards option number 1, hence this issue.

```
<__tim> jpakkane, sorry if've asked this before, but why can't add_global_*() be used with subprojects? Why can't we just make it work by using that as extra subproject-global prepend args?
<jpakkane> __tim: because subprojects must never be able to push any state to the outside. It may only be explicitly pulled.
<jpakkane> We probably could add add_project_arguments, though.
<__tim> yeah, agree. What I was suggesting is basically that add_global_arguments() in a subproject gets automatically translated to sth like add_project_arguments()
<jpakkane> No because that would be confusing. Unambiguous names are important.
<__tim> ok
<__tim> I think doing the only sane expected thing is also important :)
<jpakkane> Reducing the cognitive load on users is what we are all about. Otherwise people could just use Autotools.
<__tim> let's add add_project_arguments then()
<__tim> jpakkane, cheap ;)
<jpakkane> Sometimes you need to go there.
<__tim> once we have both the rest is a matter of documentation ;)
<jpakkane> Yep.
```

**Comment by nirbheek** [comment]
Yes, I think `project` is a better descriptor than `global` and we're young enough that the deprecation and transition would not be painful.

**Comment by arteymix** [comment]
Got to say that I fully agree here and I would be glad to make the transition in my current projects.

Global arguments should exclusively come from either `-D<lang>_args` or `<lang>FLAGS` environment variables. More generally, any project should have the possibility to become a subproject with proper isolation ensured.

**Comment by thiblahute** [comment]
https://github.com/mesonbuild/meson/pull/1030

**Comment by thiblahute** [comment]
Any way to get that one in 0.36?

**Comment by jpakkane** [comment]
It is the next one in my review queue.

**Comment by nirbheek** [comment]
Fixed via #1030
